umh-core/pkg/service/filesystem/buffered_filesystem.go

```
@@ -0,0 +1,1450 @@
+
+					if isIncremental && job.cf.Info.Size() > previousSize {
+						// Incremental read - only read the new bytes
+						file, err := os.Open(job.absPath)
+						if err != nil {
+							results <- fileReadResult{
+								absPath: job.absPath,
+								err:     fmt.Errorf("failed to open file for incremental read: %w", err),
+							}
+							continue
+						}
+						defer file.Close()
```

golangci-lint (1.64.8)

SA9001: defers in this range loop won't run unless the channel gets closed

(staticcheck)